### ISSUE TITLE
%AppData% folder no replace by path wix4

### ISSUE THREAD

**Issue by grizoood** [oleg-shilo/wixsharp#1503]
When I set the directory to: 

```c#
var project = new ManagedProject("MyProduct",
                  new Dir(@"%AppData%\My Company\My Product",
                       new File("Program.cs")));
```
or
```c#
var project = new ManagedProject("MyProduct",
                  new Dir(@"%CommonAppDataFolder%\My Company\My Product",
                       new File("Program.cs")));
```

I get this in the InstallDirDialog WPF window :
![image](https://github.com/oleg-shilo/wixsharp/assets/25281140/ee11fa08-7426-458f-ad00-4190ba148d73)

It does not replace the path with the full path.

%AppData% => C:\Users\USER_NAME\AppData\Roaming

This works: %ProgramFiles%

Here is the full example
```c#
static void Main()
  {
      var project = new ManagedProject("MyProduct",
                        new Dir(@"%AppData%\My Company\My Product",
                            new File("Program.cs")));

      project.GUID = new Guid("6fe30b47-2577-43ad-9095-1861ba25889b");

       project.ManagedUI = ManagedUI.DefaultWpf; // all stock UI dialogs

      //custom set of UI WPF dialogs
      project.ManagedUI = new ManagedUI();

      project.ManagedUI.InstallDialogs.Add<WelcomeDialog>()
                                      .Add<InstallDirDialog>()
                                      .Add<ProgressDialog>()
                                      .Add<ExitDialog>();

      project.ManagedUI.ModifyDialogs.Add<MaintenanceTypeDialog>()
                                     .Add<FeaturesDialog>()
                                     .Add<ProgressDialog>()
                                     .Add<ExitDialog>();

      project.BuildMsi();
  }
```

**Comment by grizoood** [comment]
```c#
    public string GetDirectoryPath(string name)
    {
        string[] value = (from x in GetDirectoryPathParts(name)
                          select x.AsWixVarToPath()).ToArray();
        return string.Join("\\", value);
    }
```

![image](https://github.com/oleg-shilo/wixsharp/assets/25281140/6aa33467-4458-42ef-9202-48fbc78d0aaa)

![image](https://github.com/oleg-shilo/wixsharp/assets/25281140/d4f3dd67-b9a5-41e3-90a5-a7d7df7fe5ed)

GetDirectoryPath of MsiParser return : "AppData\My Company\My Product".

AsWixVarToPath of the Extensions class in WixSharp does not contain the "AppData" string.

**Comment by oleg-shilo** [comment]
Thank you for reporting this. Indeed it is a problem. 
It is caused by the fact that the way directories are mapped is changed from WiX3 and WiX4. After the change WixSharp cannot map AppData as it expected `AppDataFolder` (as it was in WiX3).

This is how the Directory table looks: 
![image](https://github.com/oleg-shilo/wixsharp/assets/16729806/09bc6941-c759-4f33-a75a-ef75189215b1)

I discovered it only because your code works just fine in WiX3 but not in WiX4.

I have updated the WixSharp algorithm. The fix will be available in the very next release.I
